### PR TITLE
Update version of common packages

### PR DESCRIPTION
**Pull request by utpilla**
## Changes
- Update version of common packages

### CODE REVIEW

**Review by Kielek — APPROVED** [review]

File: Directory.Packages.props [hunk 1]
```diff
@@ -12,11 +12,11 @@
     <PackageVersion Include="Google.Protobuf" Version="[3.19.4,4.0)" />
     <PackageVersion Include="Grpc" Version="[2.44.0,3.0)" />
     <PackageVersion Include="Grpc.Net.Client" Version="[2.52.0,3.0)" />
-    <PackageVersion Include="Grpc.Tools" Version="[2.44.0,3.0)" />
+    <PackageVersion Include="Grpc.Tools" Version="[2.56.2,3.0)" />
     <PackageVersion Include="Microsoft.AspNetCore.Http.Abstractions" Version="[2.1.1,6.0)" />
     <PackageVersion Include="Microsoft.AspNetCore.Http.Features" Version="[2.1.1,6.0)" />
     <PackageVersion Include="Microsoft.CodeAnalysis.PublicApiAnalyzers" Version="[3.3.3]" />
-    <PackageVersion Include="Microsoft.CodeCoverage" Version="[17.4.1]" />
+    <PackageVersion Include="Microsoft.CodeCoverage" Version="[17.6.3]" />
```
Comment (Kielek) [inline Directory.Packages.props:19]: I think you should change the powershell script used by code coverage. Internal path was changed

Comment (utpilla) [reply Directory.Packages.props]: That shouldn't be the case as the script does not look for a hard-coded path.

File: Directory.Packages.props [hunk 2]
```diff
@@ -12,7 +12,7 @@
     <PackageVersion Include="Google.Protobuf" Version="[3.19.4,4.0)" />
     <PackageVersion Include="Grpc" Version="[2.44.0,3.0)" />
     <PackageVersion Include="Grpc.Net.Client" Version="[2.52.0,3.0)" />
-    <PackageVersion Include="Grpc.Tools" Version="[2.44.0,3.0)" />
+    <PackageVersion Include="Grpc.Tools" Version="[2.56.2,3.0)" />
```
Comment (reyang) [inline Directory.Packages.props:15]: Just making sure we have clarity, is this considered "prod package" and why? Why would we want to update the version?

Comment (utpilla) [reply Directory.Packages.props:15]: OTLP Exporter consumes `Grpc.Tools` package but it's marked with `PrivateAssets="all"` so it doesn't show up as a package dependency.

> Why would we want to update the version?

`Grpc.Tools` package doesn't have a dependency on any other packages. It wouldn't hurt to update the version and get its latest changes considering its usage is more of development harness.

File: Directory.Packages.props [hunk 3]
```diff
@@ -60,26 +60,26 @@
...
+    <PackageVersion Include="BenchmarkDotNet" Version="[0.13.6,0.14)" />
     <PackageVersion Include="CommandLineParser" Version="[2.9.1,3.0)" />
-    <PackageVersion Include="Grpc.AspNetCore" Version="[2.50.0,3.0)" />
-    <PackageVersion Include="Grpc.AspNetCore.Server" Version="[2.48.0, 3.0)" />
-    <PackageVersion Include="Microsoft.AspNetCore.TestHost" Version="7.0.0" />
-    <PackageVersion Include="Microsoft.AspNetCore.Mvc.Testing" Version="7.0.1" />
+    <PackageVersion Include="Grpc.AspNetCore" Version="[2.55.0,3.0)" />
+    <PackageVersion Include="Grpc.AspNetCore.Server" Version="[2.55.0, 3.0)" />
+    <PackageVersion Include="Microsoft.AspNetCore.TestHost" Version="7.0.9" />
+    <PackageVersion Include="Microsoft.AspNetCore.Mvc.Testing" Version="7.0.9" />
```
Comment (vishweshbankwar) [inline Directory.Packages.props:68]: Not sure if we could just use the single version here https://github.com/open-telemetry/opentelemetry-dotnet/blob/a76564e33bc69e796f672544c0d0e23090dc587c/test/OpenTelemetry.Instrumentation.AspNetCore.Tests/OpenTelemetry.Instrumentation.AspNetCore.Tests.csproj#L33-L39. (can be checked outside of this PR)

File: Directory.Packages.props (hunk 2 above)
Comment (reyang) [reply Directory.Packages.props:15]: Got it, sounds like this package needs to be moved to nonprod?

Comment (reyang) [reply Directory.Packages.props:15]: Doesn't block this PR though.

**Review by reyang — APPROVED** [review]

**Pull request merged**